src/atomicdex/managers/addressbook.manager.cpp

```diff
@@ -0,0 +1,303 @@
+    }
+    
+    nlohmann::json& addressbook_manager::get_contacts() noexcept
+    {
+        return data();
+    }
+    
+    const nlohmann::json& addressbook_manager::get_contact(const std::string& name) const
+    {
+        for (auto it = m_data.begin(); it != m_data.end(); ++it)
+        {
+            if (it.value().at("name") == name)
+            {
+                return it.value();
+            }
+        }
+        throw std::invalid_argument("(addressbook_manager::get_contact) given contact name does not exist");
+    }
+    
+    nlohmann::json& addressbook_manager::get_contact(const std::string& name)
+    {
+        return const_cast<nlohmann::json&>(std::as_const(*this).get_contact(name));
+    }
+    
+    const nlohmann::json& addressbook_manager::get_wallets_info(const std::string& name) const
+    {
+        return get_contact(name).at("wallets_info");
+    }
+    
+    nlohmann::json& addressbook_manager::get_wallets_info(const std::string& name)
+    {
+        return const_cast<nlohmann::json&>(std::as_const(*this).get_wallets_info(name));
+    }
+    
+    const nlohmann::json& addressbook_manager::get_wallet_info(const std::string& name, const std::string& type) const
+    {
+        for (const auto& wallet_info : get_wallets_info(name))
+        {
+            if (wallet_info.at("type").get<std::string>() == type)
+            {
+                return wallet_info;
+            }
+        }
+        throw std::invalid_argument("(addressbook_manager::get_wallet_info) given wallet info type does not exist");
+    }
+    
+    nlohmann::json& addressbook_manager::get_wallet_info(const std::string& name, const std::string& type)
+    {
+        return const_cast<nlohmann::json&>(std::as_const(*this).get_wallet_info(name, type));
+    }
+    
+    const nlohmann::json& addressbook_manager::get_wallet_info_address(const std::string& name, const std::string& type, const std::string& key) const
+    {
+        return get_wallet_info(name, type).at("addresses").at(key);
+    }
+    
+    nlohmann::json& addressbook_manager::get_wallet_info_address(const std::string& name, const std::string& type, const std::string& key)
+    {
+        return const_cast<nlohmann::json&>(std::as_const(*this).get_wallet_info_address(name, type, key));
+    }
+    
+    const nlohmann::json& addressbook_manager::get_categories(const std::string& name) const
+    {
+        return get_contact(name).at("categories");
+    }
+    
+    nlohmann::json& addressbook_manager::get_categories(const std::string& name)
+    {
+        return const_cast<nlohmann::json&>(std::as_const(*this).get_categories(name));
+    }
+}
+
+//! Modifiers
+namespace atomic_dex
+{
+    void addressbook_manager::add_contact(const std::string& name)
+    {
+        nlohmann::json contact = nlohmann::json::object();
+    
+        contact["name"] = name;
+        contact["categories"] = nlohmann::json::array();
+        contact["wallets_info"] = nlohmann::json::array();
```

Comment:
You can achieve the same this way:

```cpp
json j2 = {
  {"pi", 3.141},
  {"happy", true},
  {"name", "Niels"},
  {"nothing", nullptr},
  {"answer", {
    {"everything", 42}
  }},
  {"list", {1, 0, 2}},
  {"object", {
    {"currency", "USD"},
    {"value", 42.99}
  }}
};
```

This is more efficient, and a single initialization